src/main/java/io/aiven/kafka/connect/gcs/TopicPartitionRecordGrouper.java

```diff
@@ -96,11 +106,18 @@ public void put(final SinkRecord record) {
     }
 
     private String renderFilename(final TopicPartition tp, final SinkRecord headRecord) {
+
         return filenameTemplate.instance()
             .bindVariable(FilenameTemplateVariable.TOPIC.name, tp::topic)
-            .bindVariable(FilenameTemplateVariable.PARTITION.name, () -> Integer.toString(tp.partition()))
-            .bindVariable(FilenameTemplateVariable.START_OFFSET.name, () -> Long.toString(headRecord.kafkaOffset()))
-            .render();
+            .bindVariable(
+                FilenameTemplateVariable.PARTITION.name,
+                () -> Integer.toString(tp.partition())
+            ).bindVariable(
+                FilenameTemplateVariable.START_OFFSET.name,
+                usePaddingParameter -> usePaddingParameter.asBoolean()
+                    ? String.format("%010d", headRecord.kafkaOffset())
```

Comment:
Let's stick to 20, which is enough to accommodate even the largest offsets? [Here's how it's done in Kafka](https://github.com/apache/kafka/blob/b13a7ff7077f05b9dd21a3cd21fed6d0da5b8cdd/core/src/main/scala/kafka/log/Log.scala#L2477).